src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -1,74 +1,98 @@
-
-            m_motor.restoreFactoryDefaults();
-            m_motor.setSmartCurrentLimit(70);
-            m_pidController = m_motor.getPIDController();
+        m_motor.setCANTimeout(500);//Units in miliseconds
+        m_motor.setIdleMode(IdleMode.kCoast);
+        m_motor.setInverted(false);
 
-            m_encoder = m_motor.getEncoder();
+        m_pidController = m_motor.getPIDController();
+        m_encoder = m_motor.getEncoder();   
+        m_encoder.setAverageDepth(1);//check if 2 would work better
 
-            //confi
-            kP = 6e-5;
-            kD = 0;
-            kFF = 0.000015;
-            kMaxOutput = 1;
-            kMinOutput = -1;
-            maxRPM = 5700; 
+        //Voltage compensation
+        m_motor.enableVoltageCompensation(12); //adjust on final robot
+        m_motor.setSmartCurrentLimit(40);  
+        setBrake(false);
 
-            m_pidController.setP(kP);
-            m_pidController.setD(kD);
-            m_pidController.setFF(kFF);
-            m_pidController.setOutputRange(kMaxOutput, kMinOutput);
-        }
+        m_pidController.setOutputRange(Config.ShooterConstants.kMinOutput, Config.ShooterConstants.kMaxOutput);
+        setPIDGains(kP.get(), kI.get(), kD.get());
+        setFFGains(kFF.get());
 
-        public double getVelocity() {
+        NetworkTable shooterTable = NetworkTableInstance.getDefault().getTable("Shooter");
+        velocityPub = shooterTable.getDoubleTopic("Shooter Velocity RPM").publish(PubSubOption.periodic(0.02));
+    }
 
-                return m_encoder.getVelocity();
+    public double getVelocityRPM() {
+        return m_encoder.getVelocity();
+    }
 
-        }
+    public void setRPM(double setPoint) {
+        m_pidController.setReference(setPoint, ControlType.kVelocity);
+    }
 
-        public void setSetPoint(double setPoint){
+    public void setVoltage(double setVolt) {
+        m_motor.setVoltage(setVolt);
+    }
 
-            
-            m_pidController.setReference(setPoint, ControlType.kVelocity);
+    public void setBrake(boolean enableBreak){
+        m_motor.setIdleMode(enableBreak ? IdleMode.kBrake: IdleMode.kCoast);
+    }
 
-        }
+    private void setPIDGains(double kP, double kI, double kD){
+        m_pidController.setP(kP);
+        m_pidController.setI(kI);
+        m_pidController.setD(kD);
+    }   
 
-        public void setSetVolt(double setVolt){
-            
-           m_motor.setVoltage(setVolt);
+    private void setFFGains(double kFF){
+        m_pidController.setFF(kFF);
+    }   
 
-           //need to add a delay so the shooter can spin up, then the intake will lunch the note up into the shoorter part.
-            
-        }
+    @Override
+    public void periodic() {
+        TunableNumber.ifChanged(hashCode(), ()->setPIDGains(kP.get(), kI.get(), kD.get()), kP, kI, kD);
+        TunableNumber.ifChanged(hashCode(), ()->setFFGains(kFF.get()), kFF);
 
+        velocityPub.accept(getVelocityRPM());
+    }
```

Comment:
We should have one method (e.g. stop()) to stopMotor

Reply:
Done